app/javascript/react/src/containers/LocationForm.js

```diff
@@ -14,35 +15,143 @@ class LocationForm extends Component {
       city: "",
       state: "",
       zip: "",
-      url: ""
+      url: "",
+      errorObj: {
+        name: this.errorDictionary('name', '').message,
+        description: this.errorDictionary('description', '').message,
+        address: this.errorDictionary('address', '').message,
+        city: this.errorDictionary('city', '').message,
+        state: this.errorDictionary('state', '').message,
+        zip: this.errorDictionary('zip', '').message,
+        url: this.errorDictionary('url', '').message
+      }
     }
 
     this.handleChange = this.handleChange.bind(this);
     this.handleFormSubmit = this.handleFormSubmit.bind(this);
     this.handleClearForm = this.handleClearForm.bind(this);
+    this.errorHandler = this.errorHandler.bind(this);
   }
 
   handleChange(event) {
-    this.setState({ [event.target.name]: event.target.value })
+    this.errorHandler(event.target.name, event.target.value)
+    this.setState({ [event.target.name]: event.target.value})
+  }
+
```

Comment:
This file is getting suuuuuuuuuuuper long. Once you get your tests going, it's probably a good idea to refactor the logic and break it out into smaller components!